core/state.go

```diff
@@ -94,18 +94,19 @@ func (s *State) Root() (*felt.Felt, error) {
 
 // storage returns a [core.Trie] that represents the Starknet global state in the given Txn context.
 func (s *State) storage() (*trie.Trie, func() error, error) {
-	return s.globalTrie(db.StateTrie, db.StateRootKey, trie.NewTriePedersen)
+	return s.globalTrie(db.StateTrie, trie.NewTriePedersen)
 }
 
 func (s *State) classesTrie() (*trie.Trie, func() error, error) {
-	return s.globalTrie(db.ClassesTrie, db.ClassesRootKey, trie.NewTriePoseidon)
+	return s.globalTrie(db.ClassesTrie, trie.NewTriePoseidon)
 }
 
-func (s *State) globalTrie(bucket, rootKeyBucket db.Bucket, newTrie trie.NewTrieFunc) (*trie.Trie, func() error, error) {
-	tTxn := NewTransactionStorage(s.txn, bucket.Key())
+func (s *State) globalTrie(bucket db.Bucket, newTrie trie.NewTrieFunc) (*trie.Trie, func() error, error) {
```

Comment:
We could as well just pass in the key also instead of the whole bucket
```suggestion
func (s *State) globalTrie(key []byte, newTrie trie.NewTrieFunc) (*trie.Trie, func() error, error) {
```

Reply:
We shouldn't allow any arbitrary key. If you need a new global Trie, you shouldnt be able to do that without creating a new bucket for it. Else you might corrupt the DB by accidentally using an existing bucket, that is why we maintain the bucket list

Reply:
Got it. Thanks